src/lib/util/statistics.ts

```diff
@@ -0,0 +1,371 @@
+    | 'scale_max'
+    | 'scale_min'
+  >
+  results: StringResult
+}
+
+export interface NumberTypeQuestion {
+  type: 'Number'
+  question: Omit<
+    QuestionDetails,
+    | 'questionnaireID'
+    | 'options'
+    | 'regex_pattern'
+    | 'scale_label_left'
+    | 'scale_label_right'
+    | 'scale_max'
+    | 'scale_min'
+  >
+  results: StringResult
+}
+
+export interface MultipleChoiceTypeQuestion {
+  type: 'MultipleChoice'
+  question: Omit<
+    QuestionDetails,
+    | 'questionnaireID'
+    | 'max_bound'
+    | 'min_bound'
+    | 'regex_pattern'
+    | 'scale_label_left'
+    | 'scale_label_right'
+    | 'scale_max'
+    | 'scale_min'
+  >
+  results: ArrayResult
+}
+
+export interface CheckboxTypeQuestion {
+  type: 'Checkbox'
+  question: Omit<
+    QuestionDetails,
+    | 'questionnaireID'
+    | 'max_bound'
+    | 'min_bound'
+    | 'regex_pattern'
+    | 'scale_label_left'
+    | 'scale_label_right'
+    | 'scale_max'
+    | 'scale_min'
+  >
+  results: ArrayResult
+}
+
+export interface LinearScaleTypeQuestion {
+  type: 'LinearScale'
+  question: Omit<
+    QuestionDetails,
+    'questionnaireID' | 'max_bound' | 'min_bound' | 'options' | 'regex_pattern'
+  >
+  results: StringResult
+}
+
+export interface ResonsePerQuestionWithUser extends ResponseBody {
+  submitted_at: string
+  modified_at: string
+  traqID: string
+}
+
+export interface ResultsPerQuestion {
+  questionnaire: QuestionnaireByID
+  questions: AllTypeQuestionUnion[]
+}
+
+export type AllTypeQuestionUnion =
+  | TextTypeQuestion
+  | TextAreaTypeQuestion
+  | MultipleChoiceTypeQuestion
+  | NumberTypeQuestion
+  | CheckboxTypeQuestion
+  | LinearScaleTypeQuestion
+
+export interface BaseTypeQuestion {
+  type: QuestionTypeUnion
+  question: Partial<QuestionDetails>
+  results: ResonsePerQuestionWithUser[]
+}
+
+export type QuestionTypeUnion =
+  | 'Text'
+  | 'TextArea'
+  | 'Number'
+  | 'MultipleChoice'
+  | 'Checkbox'
+  | 'LinearScale'
+
+export const isValidTypeQuestion = (
+  question: BaseTypeQuestion
+): question is AllTypeQuestionUnion =>
+  [
+    'Text',
+    'TextArea',
+    'Number',
+    'MultipleChoice',
+    'Checkbox',
+    'LinearScale'
+  ].includes(question.type)
+
+//isSelectTypeの型ガード版
+export const isArrayQuestion = (
+  question: AllTypeQuestionUnion
+): question is CheckboxTypeQuestion | MultipleChoiceTypeQuestion =>
+  ['MultipleChoice', 'Checkbox', 'Dropdown'].includes(question.type)
+
+//isNumberTypeの型ガード版
+export const isNumberQuestion = (
+  question: AllTypeQuestionUnion
+): question is NumberTypeQuestion | LinearScaleTypeQuestion =>
+  ['LinearScale', 'Number'].includes(question.type)
+
+//['MultipleChoice', 'Checkbox', 'Dropdown'
+export type ArrayResult = (Omit<ResonsePerQuestionWithUser, 'response'> & {
+  option_response: string
+})[]
+
+//'text'とか,'LinearScale', 'Number'は数値にしてOK
+export type StringResult = (Omit<
+  ResonsePerQuestionWithUser,
+  'option_response'
+> & { response: string })[]
+
+export const generateIdTable = (
```

Comment:
```suggestion
export const generateChoiceIdsArray = (
```
かな
別にテーブル作ってるわけじゃないし